src/vidore_benchmark/retrievers/dummy_retriever.py

```diff
@@ -14,34 +13,45 @@
 @register_vision_retriever("dummy_retriever")
 class DummyRetriever(VisionRetriever):
     """
-    Dummy retriever for testing purposes. It generates random embeddings and scores.
-
-    NOTE: The dummy retriever takes PIL images in its `forward_documents` method.
+    Dummy retriever that generates random dense embeddings.
     """
 
     def __init__(
         self,
         emb_dim_query: int = 16,
         emb_dim_doc: int = 16,
+        device: str = "cpu",
     ):
         super().__init__()
+
         self.emb_dim_query = emb_dim_query
         self.emb_dim_doc = emb_dim_doc
+        self.device = device
 
     @property
     def use_visual_embedding(self) -> bool:
         return True
 
-    def forward_queries(self, queries: List[str], batch_size: int, **kwargs) -> List[Tensor]:
-        return [torch.randn(batch_size, self.emb_dim_query) for _ in range(math.ceil(len(queries) / batch_size))]
+    def forward_queries(self, queries: List[str], batch_size: int, **kwargs) -> Tensor:
+        return torch.randn(len(queries), self.emb_dim_query).to(self.device)
```

Comment:
why the torch randn ? was it just to mock ?

Reply:
`torch.randn` is used as part of the the `dummy_retriever` model class, and indeed I'm using it in my `test_evaluate_retriever.py` which tests the `evaluate-retriever` CLI command on the [`vidore/syntheticDocQA_dummy`](https://huggingface.co/datasets/vidore/syntheticDocQA_dummy) dummy dataset. While I could have mocked another of our retrievers, I decided having a dummy retriever was simpler.

The exact command that is run in this test is:

```bash
vidore-benchmark evaluate-retriever \
    --model-class dummy_retriever \
    --dataset-name vidore/syntheticDocQA_dummy \
    --split test \
    --batch-query 2 \
    --batch-passage 2 \
    --batch-score 2
```